include/vara/Feature/Constraint.h

```diff
@@ -25,55 +25,166 @@ class Constraint {
   ConstraintKind Kind;
 };
 
-class ConstraintOperator {
-public:
-  /// TODO: Numeric Operators are missing (e.g. GREATER, LESS, MINUS, ADD, MULTIPLY, DIVISION)
-  enum class OperatorKind { OK_NOT, OK_NEGATION, OK_AND, OK_OR, OK_XOR, OK_EQUALS, OK_IMPLIES, OK_EQUIVALENCE };
-  ConstraintOperator(OperatorKind Operator) : Operator(Operator) {};
-
-  [[nodiscard]] std::string toString() {
-    switch(Operator) {
-    case OperatorKind::OK_NOT: return "!";
-    case OperatorKind::OK_NEGATION: return "~";
-    case OperatorKind::OK_AND: return "&";
-    case OperatorKind::OK_OR: return "v";
-    case OperatorKind::OK_EQUALS: return "=";
-    case OperatorKind::OK_IMPLIES: return "=>";
-    case OperatorKind::OK_EQUIVALENCE: return "<=>";
-    case OperatorKind::OK_XOR: return "^";
-    default: return "";
-    }
-  };
+class BoolschesConstraint {
+public:
+  [[nodiscard]] virtual std::string toString() const;
+};
 
-private:
-  OperatorKind Operator;
+class NumericConstraint {
+public:
+  [[nodiscard]] virtual std::string toString() const;
 };
 
 class BinaryConstraint : public Constraint {
 public:
-  BinaryConstraint(ConstraintOperator Operator, Constraint *LeftOperand,
-                   Constraint *RightOperand)
-      : Operator(Operator), LeftOperand(LeftOperand),
+  BinaryConstraint(Constraint *LeftOperand, Constraint *RightOperand)
+      : LeftOperand(LeftOperand),
         RightOperand(RightOperand), Constraint(ConstraintKind::CK_BINARY) {}
 
+  [[nodiscard]] virtual std::string toString() const;
```

Comment:
```suggestion
  [[nodiscard]] std::string toString() const override;
```

Reply:
If i change this to 
```suggestion
  [[nodiscard]] virtual std::string toString() const;
```
there is no virtual function, which means the class is not abstract, right?